net-misc/openssh/patches/openssh-9.6p1.patchset

```diff
@@ -131,72 +129,30 @@ index 4f2ad8f..8946d87 100644
  	memcpy(digest, context->state.st64, SHA384_DIGEST_LENGTH);
  #endif
 -- 
-2.37.3
-
-
-From 2364d47154217df24ec1474099a87cf58a57621b Mon Sep 17 00:00:00 2001
-From: Adrien Destugues <pulkomandy@pulkomandy.tk>
-Date: Thu, 16 Jul 2020 17:57:38 +0200
-Subject: applying patch pkcs11-gcc2-build-fix.patch
-
-
-diff --git a/ssh-pkcs11-client.c b/ssh-pkcs11-client.c
-index cfd833d..9def20d 100644
---- a/ssh-pkcs11-client.c
-+++ b/ssh-pkcs11-client.c
-@@ -248,6 +248,7 @@ pkcs11_start_helper_methods(void)
- 		return (0);
- 
- #if defined(OPENSSL_HAS_ECC) && defined(HAVE_EC_KEY_METHOD_NEW)
-+{
- 	int (*orig_sign)(int, const unsigned char *, int, unsigned char *,
- 	    unsigned int *, const BIGNUM *, const BIGNUM *, EC_KEY *) = NULL;
- 	if (helper_ecdsa != NULL)
-@@ -257,6 +258,7 @@ pkcs11_start_helper_methods(void)
- 		return (-1);
- 	EC_KEY_METHOD_get_sign(helper_ecdsa, &orig_sign, NULL, NULL);
- 	EC_KEY_METHOD_set_sign(helper_ecdsa, orig_sign, NULL, ecdsa_do_sign);
-+}
- #endif /* OPENSSL_HAS_ECC && HAVE_EC_KEY_METHOD_NEW */
- 
- 	if ((helper_rsa = RSA_meth_dup(RSA_get_default_method())) == NULL)
--- 
-2.37.3
+2.42.1
 
 
-From ee7e0513d2facdae338b96786b68e500ee298e53 Mon Sep 17 00:00:00 2001
+From 48320a6881fbe9a57493a4ac86d306223b028333 Mon Sep 17 00:00:00 2001
 From: Adrien Destugues <pulkomandy@pulkomandy.tk>
 Date: Thu, 16 Jul 2020 18:08:27 +0200
 Subject: Fix configuration path in manpages
 
 
-diff --git a/contrib/ssh-copy-id.1 b/contrib/ssh-copy-id.1
-index c141a29..fd27f1d 100644
---- a/contrib/ssh-copy-id.1
-+++ b/contrib/ssh-copy-id.1
-@@ -54,7 +54,7 @@ this may result in you being repeatedly prompted for pass-phrases).
- It then assembles a list of those that failed to log in, and using ssh,
- enables logins with those keys on the remote server.  By default it adds
- the keys by appending them to the remote user's
--.Pa ~/.ssh/authorized_keys
-+.Pa ~/config/settings/settings/ssh/authorized_keys
- (creating the file, and directory, if necessary).  It is also capable
- of detecting if the remote system is a NetScreen, and using its
- .Ql set ssh pka-dsa key ...
-@@ -124,9 +124,9 @@ will be used.
- The
- .Ic default_ID_file
- is the most recent file that matches:
--.Pa ~/.ssh/id*.pub ,
-+.Pa ~/config/settings/settings/ssh/id*.pub ,
- (excluding those that match
--.Pa ~/.ssh/*-cert.pub )
-+.Pa ~/config/settings/settings/ssh/*-cert.pub )
- so if you create a key that is not the one you want
- .Nm
- to use, just use
+diff --git a/contrib/ssh-copy-id b/contrib/ssh-copy-id
```

Comment:
this doesn't look correct. The patch should be applied to contrib/ssh-copy-id.1

Reply:
jikes, thanks for spotting!

Reply:
Fixed, in ssh-copy-id there is still `TARGET_PATH=".ssh/authorized_keys"` shouldn't that be changed too?

Reply:
> Fixed, in ssh-copy-id there is still `TARGET_PATH=".ssh/authorized_keys"` shouldn't that be changed too?

The target of ssh-copy-id can be anything, not just Haiku, and for other unix-like systems, this is correct. Ideally, it would have to detect what the target system is and adjust the path accordingly.

This is also a problem when trying to use it from Linux for a Haiku system, so that problem can't be fixed easily, unfortunately.

Reply:
Thanks, looks good